src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/ReflectionFunction.cs

```diff
@@ -270,9 +277,38 @@ public async Task<FormulaValue> InvokeAsync(IServiceProvider serviceProvider, Fo
 
             formulaResult ??= FormulaValue.NewBlank(_info.RetType);
 
-            if (!formulaResult.Type._type.Accepts(_info.RetType._type, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: true))
+            var dataSourceType = formulaResult.Type._type;

```

Comment:
>dataSourceType [](http://example.com/codeflow?start=16&length=14)

This doesn't have to be a data source, we can have any type here. Naming it as such can be misleading

Reply:
update name